src/pages/account/story/StoryEdit.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect } from "react";
+import { Redirect, useParams } from "react-router-dom";
+
+import axios from "axios";
+import axiosApi from "../../../utils/AxiosApi";
+import FieldSet from "../../../components/Form/Fieldset";
+import Input from "../../../components/Form/Input";
+import TextArea from "../../../components/Form/TextArea";
+import Button from "../../../components/Button";
+import NavBar from "../../../components/NavBar";
+
+const StoryEdit = () => {
+  const [input, setInput] = useState({});
+  const [story, setStory] = useState();
+  const [loading, setLoading] = useState(false);
+  const [redirect, setRedirect] = useState(false);
+  const [mediaPreview, setMediaPreview] = useState("");
+
+  const { storyId } = useParams();
+
+  useEffect(() => {
+    const fetchStoryData = async () => {
+      const result = await axiosApi.get(`/storyDetails/${storyId}`);
+      const singleStory = await result.data;
+      setStory(singleStory);
+    };
+    fetchStoryData();
+  }, [storyId]);
+
+  const handleImageUpload = async () => {
+    const data = new FormData();
+    data.append("file", input.media);
+    data.append("upload_preset", "idnmaxun");
+    const response = await axios.post(
+      process.env.REACT_APP_CLOUDINARY_URL,
+      data
+    );
+    const mediaUrl = await response.data.url;
+    input.media = mediaUrl;
+  };
+
+  const handleChange = (event) => {
+    const { name, value, files } = event.target;
+    if (name === "media") {
+      setInput((prevState) => ({ ...prevState, media: files[0] }));
+      setMediaPreview(window.URL.createObjectURL(files[0]));
```

Comment:
What does this `window.URL.createObjectURL` do and why do you need it for?
I actually prefer you don't use the `window` object in React, because that means you're stepping outside of the world of React which can cause weird side effects. So only you really know what you're doing you can do it

Reply:
I'm using it to show a preview of the picture you're about to upload.